docs/architecture/maui/unit-testing.md

```diff
@@ -1,9 +1,223 @@
+description: Providing application testing and improving code quality
 author: michaelstonis
 no-loc: [MAUI]
-ms.date: 07/12/2022
+ms.date: 07/11/2022
 ---
 
-# Unit Testing
+# Unit testing
+
+Mobile apps have unique problems that desktop and web-based applications don't typically encounter. Mobile users will differ by their devices, network connectivity, availability of services, and various other factors. Therefore, mobile apps should be tested as they would be used in the real world to improve their quality, reliability, and performance. Many types of testing should be performed on an app, including unit testing, integration testing, and user interface testing. Unit testing is the most common form of testing and is essential to building high-quality applications.
+
+A unit test takes a small unit of the app, typically a method, isolates it from the remainder of the code, and verifies that it behaves as expected. Its goal is to check that each unit of functionality performs as expected, so errors don't propagate throughout the app. Detecting a bug where it occurs is more efficient that observing the effect of a bug indirectly at a secondary point of failure.
+
+Unit testing has the most significant effect on code quality when it's an integral part of the software development workflow. Unit tests can act as design documentation and functional specifications for an application. As soon as a method has been written, unit tests should be written that verify the method's behavior in response to standard, boundary, and incorrect input data cases and check any explicit or implicit assumptions made by the code. Alternatively, with test-driven development, unit tests are written before the code. For more information on test-driven development and how to implement it, see [Walkthrough: Test-driven development using Test Explorer.](/visualstudio/test/quick-start-test-driven-development-with-test-explorer?view=vs-2022)
+
+> [!NOTE]
+> Unit tests are very effective against regression. That is, functionality that used to work, but has been disturbed by a faulty update.
+
+Unit tests typically use the arrange-act-assert pattern:
+
+| Step | Description |
+|---------|---------|
+| Arrange | Initializes objects and sets the value of the data that is passed to the method under test. |
+| Act | Invokes the method under test with the required arguments. |
+| Assert | Verifies that the action of the method under test behaves as expected. |
+
+This pattern ensures that unit tests are readable, self-describing, and consistent.
+
+## Dependency injection and unit testing
+
+One of the motivations for adopting a loosely-coupled architecture is that it facilitates unit testing. One of the types registered with the dependency injection service is the `IAppEnvironmentService` interface. The following code example shows an outline of this class:
+
+```csharp
+public class OrderDetailViewModel : ViewModelBase
+{
+    private IOrderService _ordersService;
```

Comment:
I see that you do not mention this field in the text below. But as someone who is familiar with DI and unit testing, I'm a bit befuddled by the lack of assignment to the `_ordersService` field or at least mention of its intent.

Reply:
This is entirely just a code update issue. I had updated the DI process along the way here and just missed updating the variables here. Updating references now.

Reply:
yes, I just pushed the change up.